### PR TITLE
setup: move setup to experimental dir

### PR DESCRIPTION
**Pull request by dougsland**
The experimental dir is a home to unsupported code but developers are welcome to use, test or add more code.

Setup tool is a solution to deploy QM for non automotive environment. Today the official and only solution for automotive is the ones provided via CentOS automotive sample-images project.

See: https://gitlab.com/CentOS/automotive/sample-images
See-Also: https://github.com/containers/qm/issues/430

### CODE REVIEW

**Comment by dougsland** [comment]
Okay, as expected, we are using our own food to make CI/CD. We will need to split it in a sub-package.
How about, qm-setup subpackage which is not required in the main qm package but we require this package in the CI/CD and qm-desktop? @alexlarsson @Yarboa @rhatdan @pypingou  WDYT?

Log from CI/CD error:
```
        ./set-ffi-env-e2e: line 218: /usr/share/qm/setup: No such file or directory
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by dougsland** [comment]
> Okay, as expected, we are eating our own food to make CI/CD. We will need to split it in a sub-package. How about, qm-setup subpackage which is not required in the main qm package but we require this package in the CI/CD and qm-desktop? @alexlarsson @Yarboa @rhatdan @pypingou WDYT?
> 
> Log from CI/CD error:
> 
> ```
>         ./set-ffi-env-e2e: line 218: /usr/share/qm/setup: No such file or directory
>         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> ```

or actually remake it using osbuild.

**Comment by Yarboa** [comment]
> Okay, as expected, we are eating our own food to make CI/CD. We will need to split it in a sub-package. How about, qm-setup subpackage which is not required in the main qm package but we require this package in the CI/CD and qm-desktop? @alexlarsson @Yarboa @rhatdan @pypingou WDYT?
> 
> Log from CI/CD error:
> 
> ```
>         ./set-ffi-env-e2e: line 218: /usr/share/qm/setup: No such file or directory
>         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> ```

Thanks @dougsland 
I tend to use ENV_VAR, 

So once rpm install called, if ENV_VAR exist like ADD_SETUP=true it will install it.
Default it will not, WDYT?

File: rpm/qm.spec [hunk 1]
```diff
@@ -5,7 +5,6 @@
 %global moduletype services
 %global modulenames qm
 %global seccomp_json /usr/share/%{modulenames}/seccomp.json
-%global setup_tool %{_prefix}/share/%{modulenames}/setup
```
Comment (Yarboa) [inline rpm/qm.spec:8]: I would use it here, 
%if %{include_setup} == "true"
setup_tool %{_prefix}/share/%{modulenames}/setup
%endif

And return the deleted file in Makefile

**Comment by alexlarsson** [comment]
I would much prefer it if we just kept it in /usr/share/qm/setup, but well documented as not being supported in production. Having a subpackage implies much more in terms of something being supported, and it will be very complex to ever get rid of once it is in peoples rpm databases, etc.

**Comment by dougsland** [comment]
> I would much prefer it if we just kept it in /usr/share/qm/setup, but well documented as not being supported in production. Having a subpackage implies much more in terms of something being supported, and it will be very complex to ever get rid of once it is in peoples rpm databases, etc.

You got it. Next patch coming soon.

File: rpm/qm.spec (hunk 1 above)
Comment (dougsland) [reply rpm/qm.spec:8]: I will add a patch informing setup is not the official and supported way to deploy QM for automotive as @alexlarsson shared. Will be much more simple but we can always improve on the road with `if's/env vars or subpackages` thanks @Yarboa

**Comment by dougsland** [comment]
Not this solution, we will go with an update

**Pull request closed without merging**